### PR TITLE
[dalgona] Initialize dalgona module

### PR DESCRIPTION
**Pull request by jinevening**
This adds below files to create dalgona module.
- CMakeLists.txt
- README
- Driver
- requires.cmake

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/pull/3501

### CODE REVIEW

File: compiler/dalgona/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+## Possible applications
+- Finding quantization parameters based on the distribution of activations
+- Finding sparse activations by observing the portion of zero values
+- Finding the distribution of conditional variables in If-statement and While-statement
+- Visualization of activation data with Python libraries
+
+## Prerequisite
+- Python (apt install python-dev python3-dev)
+- Circle model (target to analyze)
+- Input data of the model (hdf5 format. See rawdata2hdf5 or gen_h5_explicit_inputs.py for more details.)
```
Comment (mhs4670go) [inline compiler/dalgona/README.md:20]:
```suggestion
- Input data of the model (hdf5 format. See `rawdata2hdf5` or `gen_h5_explicit_inputs.py` for more details.)
```
Or
```suggestion
- Input data of the model (hdf5 format. See _rawdata2hdf5_ or _gen_h5_explicit_inputs.py_ for more details.)
```

File: compiler/dalgona/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+# dalgona
+
+## What is dalgona?
+
+_dalgona_ is a tool for dynamic analysis of deep neural network.
+
+## How it works?
+
+_dalgona_ runs a user's custom analysis code (written in "Python") while performing inference. The analysis code has the form of hooks, called before/after each operator is executed. Intermediate execution results (values of activations) are passed to the hooks, so users can analyze the distribution of activations inside the hooks. The analysis result can be exported as files, log messages or any other forms, used for various purposes (model compression, optimization, etc.).
```
Comment (seanshpark) [inline compiler/dalgona/README.md:9]: (optional) may be add more about by what means of `inference` is. is it with `luci-interpreter` ?

File: compiler/dalgona/README.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+## Possible applications
+- Finding quantization parameters based on the distribution of activations
+- Finding sparse activations by observing the portion of zero values
+- Finding the distribution of conditional variables in If-statement and While-statement
+- Visualization of activation data with Python libraries
+
+## Prerequisite
+- Python (apt install python-dev python3-dev)
```
Comment (seanshpark) [inline compiler/dalgona/README.md:18]: (1) maybe `sudo apt install python-dev python3-dev` ?
(2) or just `python-dev python3-dev packages`

File: compiler/dalgona/README.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+ --analysis analysis/AnalysisTemplate.py
+```
+
+## Arguments
+```
+  --help            Show help message and exit
+  --input_model     Input model filepath (.circle)
+  --input_data      Input data filepath (.h5) (if not given, random data will be used)
+  --analysis        Analysis code filepath (.py)
+  --analysis_args   (optional) String argument passed to the analysis code
```
Comment (seanshpark) [inline compiler/dalgona/README.md:37]: Q) is `--version` not supported?

**Review by mhs4670go — APPROVED** [review]
LGTM

File: compiler/dalgona/README.md (hunk 4 above)
Comment (jinevening) [reply compiler/dalgona/README.md:39]: Hmm.. I didn't think about versioning yet.  May I go as-is for now and introduce version option later when necessary?

Comment (jinevening) [reply compiler/dalgona/README.md:39]: I guess that dalgona will be used mainly by internal developers, not end-users.

File: compiler/dalgona/README.md (hunk 2 above)
Comment (jinevening) [reply compiler/dalgona/README.md:9]: Added comments.

**Review by seanshpark — APPROVED** [review]
LGMT thank you!

**Pull request merged**